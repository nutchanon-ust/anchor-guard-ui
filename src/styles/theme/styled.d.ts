import 'styled-components';
import { Theme } from './themes';

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
