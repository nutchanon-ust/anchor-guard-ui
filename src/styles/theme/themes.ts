const lightTheme = {
  primary: 'rgba(215,113,88,1)',
  text: 'rgba(58,52,51,1)',
  textSecondary: 'rgba(58,52,51,0.7)',
  background: 'rgba(255,255,255,1)',
  backgroundVariant: 'rgba(251,249,249,1)',
  border: 'rgba(58,52,51,0.12)',
  borderLight: 'rgba(58,52,51,0.05)',
};

const darkTheme: Theme = {
  primary: 'rgba(220,120,95,1)',
  text: 'rgba(241,233,231,1)',
  textSecondary: 'rgba(241,233,231,0.6)',
  background: 'rgba(0,0,0,1)',
  backgroundVariant: 'rgba(28,26,26,1)',
  border: 'rgba(241,233,231,0.15)',
  borderLight: 'rgba(241,233,231,0.05)',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
  dark: darkTheme,
};
