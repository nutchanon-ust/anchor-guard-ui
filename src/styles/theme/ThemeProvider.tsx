import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import { selectTheme } from './slice/selectors';

export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice();

  const theme = useSelector(selectTheme);
  return (
    <OriginalThemeProvider theme={theme}>
      {React.Children.only(props.children)}
    </OriginalThemeProvider>
  );
};
