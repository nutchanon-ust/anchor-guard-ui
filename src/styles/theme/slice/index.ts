import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { useInjectReducer } from 'utils/redux-injectors';
import { getThemeFromStorage } from '../utils';
import { ThemeKeyType, ThemeState } from './types';

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
};

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<ThemeKeyType>) {
      state.selected = action.payload;
    },
  },
});

export const { actions: themeActions, reducer } = slice;

export const useThemeSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer });
  return { actions: slice.actions };
};
