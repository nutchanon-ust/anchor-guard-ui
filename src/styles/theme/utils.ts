import { ThemeKeyType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeKeyType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeKeyType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeKeyType) || null
    : null;
}
