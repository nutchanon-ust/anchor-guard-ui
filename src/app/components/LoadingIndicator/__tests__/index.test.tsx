import * as React from 'react';
import { render } from '@testing-library/react';

import { LoadingIndicator } from '../index';
import { themes } from 'styles/theme/themes';
import { DefaultTheme, ThemeProvider } from 'styled-components';

const renderWithTheme = (
  props: Parameters<typeof LoadingIndicator>[number] = {},
  theme?: DefaultTheme,
) =>
  render(
    <ThemeProvider theme={theme || themes.light}>
      <LoadingIndicator {...props} />
    </ThemeProvider>,
  );

describe('<LoadingIndicator />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = renderWithTheme();
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot when props changed', () => {
    const loadingIndicator = renderWithTheme({ small: true });
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });

  it('should have theme', () => {
    const loadingIndicator = renderWithTheme();
    expect(loadingIndicator.container.querySelector('circle')).toHaveStyle(
      `stroke: ${themes.light.primary}`,
    );
  });
});
