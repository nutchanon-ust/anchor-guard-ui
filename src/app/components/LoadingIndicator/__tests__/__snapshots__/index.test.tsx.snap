// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should match snapshot 1`] = `
.c0 {
  -webkit-animation: cqkDSr 2.625s linear infinite;
  animation: cqkDSr 2.625s linear infinite;
  height: 3rem;
  width: 3rem;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c1 {
  -webkit-animation: caVMmK 1.5s ease-in-out infinite;
  animation: caVMmK 1.5s ease-in-out infinite;
  stroke: rgba(215,113,88,1);
  stroke-linecap: round;
}

<svg
  class="c0"
  viewBox="-24 -24 48 48"
>
  <circle
    class="c1"
    cx="0"
    cy="0"
    fill="none"
    r="20"
    stroke-width="4"
  />
</svg>
`;

exports[`<LoadingIndicator /> should match snapshot when props changed 1`] = `
.c0 {
  -webkit-animation: cqkDSr 2.625s linear infinite;
  animation: cqkDSr 2.625s linear infinite;
  height: 1.25rem;
  width: 1.25rem;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c1 {
  -webkit-animation: caVMmK 1.5s ease-in-out infinite;
  animation: caVMmK 1.5s ease-in-out infinite;
  stroke: rgba(215,113,88,1);
  stroke-linecap: round;
}

<svg
  class="c0"
  viewBox="-24 -24 48 48"
>
  <circle
    class="c1"
    cx="0"
    cy="0"
    fill="none"
    r="20"
    stroke-width="4"
  />
</svg>
`;
